src/Akka.Persistence.Sql/Extensions/ConnectionFactoryExtensions.cs

```diff
@@ -74,12 +95,21 @@ public static async Task<T> ExecuteWithTransactionAsync<T>(
             }
         }
         
-        internal static Task<T> ExecuteWithTransactionAsync<TState,T>(
+        internal static async Task<T> ExecuteQueryWithTransactionAsync<TState,T>(
             this DbStateHolder factory,
+            IActorRef queryPermitter,
             TState state,
             Func<AkkaDataConnection, CancellationToken, TState, Task<T>> handler)
         {
-            return factory.ConnectionFactory.ExecuteWithTransactionAsync(state, factory.IsolationLevel, factory.ShutdownToken, handler);
+            await queryPermitter.Ask<QueryStartGranted>(RequestQueryStart.Instance);
+            try
+            {
+                return await factory.ConnectionFactory.ExecuteWithTransactionAsync(state, factory.IsolationLevel, factory.ShutdownToken, handler);
+            }
+            finally
+            {
+                queryPermitter.Tell(ReturnQueryStart.Instance);
+            }
```

Comment:
This is the query throttling mechanism, we always ask for a permission before we execute any DB operations